### ISSUE TITLE
scale_regularization penalizes only the mean

### ISSUE THREAD

**Issue by lorenzobagnol** [openvdb/fvdb-reality-capture#324]
## Summary

`scale_regularization` in `GaussianSplatOptimizerMCMC` penalizes the **mean** scale across all Gaussians, which does not prevent a small number of outlier Gaussians from growing unbounded during training — even at 10x the default weight. This is relevant because oversized Gaussians directly inflate `intersect_gaussian_tiles` memory (their projected screen-space radius determines how many tiles they touch), contributing to CUDA OOM independent of `max_gaussians`.

## Environment

- `fvdb-reality-capture` version: 0.5.0
- Optimizer: `GaussianSplatOptimizerMCMCConfig`
- GPU: NVIDIA RTX A4500 (20GB)

## Mechanism

`regularization_loss()` (`gaussian_splat_optimizer_mcmc.py:358-375`) applies:

    loss = loss + self._config.scale_regularization * self._model.scales.mean()

This penalizes the **mean** scale across all Gaussians. In `_relocate()` (`gaussian_splat_optimizer_mcmc.py:448-498`), dead Gaussians (opacity below `deletion_opacity_threshold`) are relocated onto sampled "alive" donors, with a `ratios` term (`torch.bincount(sampled_idxs, ...)`) that rescales each donor's covariance in proportion to how many dead Gaussians it absorbed in that step. When many Gaussians die per refine step, a small number of donors can be sampled repeatedly and compound in scale across successive refine events.

Because `scale_regularization` only constrains the **mean**, it cannot prevent this: the bulk of the population can shrink slightly to compensate for a handful of outliers growing arbitrarily large, and the loss term is satisfied either way.

## Reproduction

Config: `scale_regularization=0.1` (10x the library default of `0.01`), all other settings otherwise standard MCMC config, run on a 50-image scene over 26 refine events (see companion issue on `opacity_regularization`/`refine_every_epoch` interaction for full run context).

    refine_count=1  scale_mean_before=0.006487  scale_max_before=0.179686
    refine_count=6  scale_mean_before=0.013519  scale_max_before=3.981085
    refine_count=13 scale_mean_before=0.013013  scale_max_before=6.359546
    refine_count=20 scale_mean_before=0.012572  scale_max_before=8.725759
    refine_count=26 scale_mean_before=0.012483  scale_max_before=15.827422

`scale_mean` stays essentially flat (~0.012-0.014) across the entire run, while `scale_max` grows ~88x (0.18 → 15.8), even at 10x the default regularization weight. This confirms `scale_regularization` cannot prevent outlier Gaussians from growing unbounded.

## Suggested fix / discussion

Consider an additional max-based or percentile-based scale regularization/clamp option (analogous to `deletion_scale_2d_threshold` / `deletion_scale_3d_threshold`, which exist on the shared `GaussianSplatOptimizerConfig` base class but appear unused by `GaussianSplatOptimizerMCMC` — grep shows no references to either in `gaussian_splat_optimizer_mcmc.py`; worth confirming whether that's intentional or an oversight). This would let runaway outliers be directly bounded rather than only indirectly discouraged on average, which the current mean-based penalty cannot do regardless of weight.
